source/slang/slang-check-decl.cpp

```diff
@@ -9974,7 +10026,7 @@ namespace Slang
                 auto stageCaps = CapabilitySet(Profile(entryPointAttr->stage).getCapabilityName());
                 if (declaredCaps.isIncompatibleWith(stageCaps))
                 {
-                    getSink()->diagnose(funcDecl->loc, Diagnostics::stageIsInCompatibleWithCapabilityDefinition, funcDecl, stageCaps, declaredCaps);
+                    getSink()->diagnoseCapabilityErrors(this->getOptionSet(), funcDecl->loc, Diagnostics::stageIsInCompatibleWithCapabilityDefinition, funcDecl, stageCaps, declaredCaps);
```

Comment:
Just make this a global function instead of extending DiagnosticSink.

This class is shared with things outside of the slang compiler so it shouldn't have any slang specific logic.